### PR TITLE
Fix #2728.

### LINKED ISSUE #2728: newline_style = "Native" + wrap_comments = true introduces extra characters into comment
The following issue seems to be limited to Windows. I'm unable to reproduce on Linux. Both `newline_style` and `wrap_comments` need to be set to cause the problem.

Starting with the following `rustfmt.toml`
```toml
newline_style = "Native"
wrap_comments = true
```

and the following `src/main.rs`
```rust
//! ```rust
//! extern crate uom;
//! ```

fn main() { }
```

running the following command
```sh
$ cargo +nightly fmt --version && cargo +nightly fmt
rustfmt 0.7.0-nightly (bf2581bf 2018-05-17)
```

adds an extra line into `src/main.rs` (note the trailing space on the new line)
```diff
diff --git a/src/main.rs b/src/main.rs
index 8bc09f5..d137f1a 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,4 +1,5 @@
 //! ```rust
+//! 
 //! extern crate uom;
 //! ```
```

The issue was original detected in a branch of [`uom`](https://github.com/iliekturtles/uom) working on `rustfmt` setup where both a new line and a `}` are inserted. I was unable to reduce this exact behavior into a minimal example, but am including the diff below for reference.

```diff
diff --git a/src/lib.rs b/src/lib.rs
index f1c54a5..f8d00b1 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -24,7 +24,9 @@
 //! and this to your crate root:
 //!
 //! ```rust
+//!
 //! extern crate uom;
+//! }
 //! ```
```

### PR DESCRIPTION
**Pull request by wada314**
Do you mind if I clone() the whole config object?

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -718,10 +718,16 @@ fn format_code_block(code_snippet: &str, config: &Config) -> Option<String> {
     let mut result = String::with_capacity(snippet.len());
     let mut is_first = true;
 
+    // While formatting the code, ignore the config's newline style setting and always use "\n"
+    // instead of "\r\n" for the newline characters.
```
Comment (nrc) [inline src/lib.rs]: Why is it OK to do this? It seems like this would insert the wrong line ending if the user wants Windows line endings.

**Comment by wada314** [comment]
We are doing '\n' ==> "\r\n' conversion for the whole output files in another place, so it should be okay. All tests are still passing in Windows after this fix.
This place is doing in-comment code block formatting. The output here is not rustfmt final output but an input to comment.rs.  We are expecting '\n' newline here.

**Comment by nrc** [comment]
Could you add that information to a comment please? r+ with that

**Comment by wada314** [comment]
Added comment. The automatic test is failing but it looks it's not related to my PR.

**Comment by nrc** [comment]
Thanks!

**Pull request merged**